modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -286,6 +284,9 @@ private static Result handleConjunction(List<Result> conjunctionsWithUnknowns) {
                     verified = false;
                 }
             }
+            // add range fields from this Result to the seenRangeFields set so that minimumShouldMatch is correctly
+            // calculated for subsequent Results
+            result.extractions.stream().filter(e -> e.range != null).forEach(e -> seenRangeFields.add(e.range.fieldName));
```

Comment:
nit: I generally have a preference for using method refs when possible and using new lines when chaining, e.g.
```suggestion
            result.extractions.stream()
                .filter(Objects::nonNull)
                .map(e -> e.range)
                .forEach(seenRangeFields::add);
```